src/origdatablocks/origdatablocks.controller.ts

```diff
@@ -102,9 +104,52 @@ export class OrigDatablocksController {
       fields: JSON.parse(filters.fields ?? "{}"),
       limits: JSON.parse(filters.limits ?? "{}"),
     };
+
     return this.origDatablocksService.fullquery(parsedFilters);
   }
 
+  // GET /origdatablocks/fullquery/files
+  @UseGuards(PoliciesGuard)
+  @CheckPolicies((ability: AppAbility) =>
+    ability.can(Action.Read, OrigDatablock),
+  )
+  @Get("/fullquery/files")
+  @ApiQuery({
+    name: "filters",
+    description:
+      "Database query and filters to apply when retrieve all origdatablocks with files",
+    required: false,
+  })
+  async fullqueryFiles(
+    @Query() filters: { fields?: string; limits?: string },
+  ): Promise<OrigDatablockFileList[] | null> {
+    const parsedFilters = {
+      fields: JSON.parse(filters.fields ?? "{}"),
+      limits: JSON.parse(filters.limits ?? "{}"),
+    };
+
+    const dataFileList: OrigDatablockFileList[] = [];
+
+    const origdatablockList = await this.origDatablocksService.fullqueryFiles(
+      parsedFilters,
+    );
+
+    // This conversion process is needed to get output directly rather than from _doc
+    const origdatablockListCopy: OrigDatablock[] | null = JSON.parse(
+      JSON.stringify(origdatablockList),
+    );
+
+    if (!origdatablockListCopy) return null;
+
+    origdatablockListCopy.forEach((data) => {
```

Comment:
I think it would be more clear to use the following:
`const dataFileList = origdatablockListCopy.map(...)`
instead of push.

Reply:
Maybe something like this will do the trick: 
`const dataFileList = origdatablockListCopy.flatMap((data) => data.dataFileList.map((file) => ({ ...data, dataFileList: file })));`